src/macaron/repo_finder/commit_finder.py

```diff
@@ -0,0 +1,509 @@
+
+    Parameters
+    ----------
+    git_obj: Git
+        The repository.
+    name: str
+        The name of the analysis target.
+    version: str
+        The version of the analysis target.
+
+    Returns
+    -------
+    tuple[str, str]
+        The branch name and digest as a tuple.
+    """
+    logger.debug("Searching for commit of artifact version using tags: %s@%s", name, version)
+
+    # Only consider tags that have a commit.
+    valid_tags = {}
+    for tag in git_obj.repo.tags:
+        commit = _get_tag_commit(tag)
+        if not commit:
+            logger.debug("No commit found for tag: %s", tag)
+            continue
+
+        tag_name = str(tag)
+        valid_tags[tag_name] = tag
+
+    if not valid_tags:
+        logger.debug("No tags with commits found for %s", name)
+        return "", ""
+
+    # Match tags.
+    matched_tags = match_tags(list(valid_tags.keys()), name, version)
+
+    if not matched_tags:
+        logger.debug("No tags matched for %s", name)
+        return "", ""
+
+    if len(matched_tags) > 1:
+        logger.debug("Tags found for %s: %s", name, len(matched_tags))
+        logger.debug("Best match: %s", matched_tags[0])
+        logger.debug("Up to 5 others: %s", matched_tags[1:6])
+
+    tag_name = matched_tags[0]
+    tag = valid_tags[tag_name]
+    if not tag:
+        # Tag names are taken from valid_tags and should always exist within it.
+        logger.debug("Missing tag name from tag dict: %s not in %s", tag_name, valid_tags.keys())
+
+    branch_name = _get_branch_of_commit(git_obj.get_commit_from_tag(tag_name))
+    if not branch_name:
+        logger.debug("No valid branch associated with tag (commit): %s (%s)", tag_name, tag.commit.hexsha)
+        return "", ""
+
+    logger.debug(
+        "Found tag %s with commit %s of branch %s for artifact version %s@%s",
+        tag,
+        tag.commit.hexsha,
+        branch_name,
+        name,
+        version,
+    )
+    return branch_name, tag.commit.hexsha
+
+
+def _build_version_pattern(name: str, version: str) -> tuple[Pattern | None, list[str], bool]:
+    """Build a version pattern to match the passed version string.
+
+    Parameters
+    ----------
+    name: str
+        The name string.
+    version: str
+        The version string.
+
+    Returns
+    -------
+    tuple[Pattern | None, list[str], bool]
+        The tuple of the regex pattern that will match the version, the list of version parts that were extracted, and
+        whether the version string has a non-numeric suffix.
+
+    """
+    # The version is split on non-alphanumeric characters to separate the version parts from the non-version parts.
+    # e.g. 1.2.3-DEV -> [1, 2, 3, DEV]
+    try:
+        split = split_pattern.split(version)
+    except TypeError as error:
+        logger.debug("Failed to perform regex split on version string %s -- %s", version, error)
+        return None, [], False
```

Comment:
Can you please add a unit test to trigger this error?

Reply:
After more testing I think the only way this can trigger is if the version is `None`. I've added a check to prevent that from happening at this stage.